flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/rules/logical/StreamLogicalWindowAggregateRule.scala

```diff
@@ -83,6 +85,14 @@ class StreamLogicalWindowAggregateRule
         throw new ValidationException("Window can only be defined over a time attribute column.")
     }
   }
+
+  def getOperandAsLong(call: RexCall, idx: Int): Long =
+    call.getOperands.get(idx) match {
+      case v: RexLiteral if v.getTypeName.getFamily == SqlTypeFamily.INTERVAL_DAY_TIME =>
+        v.getValue.asInstanceOf[JBigDecimal].longValue()
+      case _ => throw new TableException(
+        "Only constant window intervals with millisecond resolution are supported.")
```

Comment:
The exception is not clear for users. I would suggest change it to :
"Window aggregate only support SECOND, MINUTE, HOUR, DAY as the time unit. MONTH and YEAR  time unit are not supported yet." 

And keep the "Only constant window descriptors are supported" exception if it is not a literal.